### PR TITLE
Get verified state from the list

### PR DESCRIPTION
**Pull request by NejcZdovc**
Resolves https://github.com/brave/brave-browser/issues/3780

Depends on https://github.com/brave/brave-browser/issues/4025 and https://github.com/brave/brave-browser/issues/3953

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests && npm run test-security`) on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone

## Test Plan:
- clean profie
- enable rewards
- go to http://duckduckgo.com/ site
- add monthly tip
- do one time tip
- make sure that panel is verified
- make sure that you see verified check in tips section
- make sure that you added site to ac table and that you see verified check
- close the browser
- go to publisher_list and change verified status of to false: from `["duckduckgo.com",true,false,{}]` to `["duckduckgo.com",false,false,{}]`
- start the browser
- make sure that on rewards page and in the panel you see unverified status now

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [x] Verify test plan is specified in PR before merging to source

### CODE REVIEW

File: vendor/bat-native-ledger/src/bat/ledger/internal/ledger_impl.cc [hunk 1]
```diff
@@ -430,34 +438,92 @@ LedgerImpl::GetRecurringDonationPublisherInfo() {
...
+  callback(result, std::move(publisher));
+}
+
+void LedgerImpl::ModifyPublishersVerified(
+    const ledger::PublisherInfoList& list,
+    uint32_t record,
+    ledger::PublisherInfoListCallback callback) {
+  ledger::PublisherInfoList new_list;
+
+  for (const auto& publisher : list) {
```
[the remark this answers is not in the document]
Comment (NejcZdovc) [reply vendor/bat-native-ledger/src/bat/ledger/internal/ledger_impl.cc]: discussed via DM

**Comment by NejcZdovc** [comment]
@jasonrsadler like we talked we can remove it, but this needs to be a separate PR as I would need to investigate what could potentially be broken

**Comment by tmancey** [comment]
@NejcZdovc did you say there was a ticket for migrating to a database, as honestly this is going to cause us no end of problems on iOS and Android, i.e. see https://drive.google.com/file/d/0B7Vx1OvzrLa3Y0R0X1BZbUpicGc/view and we still support Android 4.4

**Comment by NejcZdovc** [comment]
@tmancey no issue yet as we didn't finalize the plan yet. We can define next week in person

**Comment by bridiver** [comment]
we can put the list in its own table and join if that works better than the field in the publisher_info table

**Comment by NejcZdovc** [comment]
yup this was my thinking as well. So this way we can remove `publisher_list` direct access completely and fetch everything from db and list just updates db every time we fetch it. So publisher banner would get data from db now as well (currently it's getting it directly from the list)

**Pull request merged**